docs/building.md

```diff
@@ -0,0 +1,176 @@
+
+## Fobnail Token firmware
+
+Start by installing prerequisites and adding current user to `docker` group:
+
+```shell
+sudo apt -y install git docker.io
+sudo usermod -aG docker $USER
+```
+
+Log out of the desktop session then log in again - this is required after making
+the change to user groups. You may confirm it by running `groups`.
+
+Clone the repository along with its submodules:
+
+```shell
+git clone https://github.com/fobnail/fobnail --recurse-submodules
+```
+
+Build process for Token firmware is by far the most complicated one. For that
+reason Fobnail SDK is strongly suggested. To use it, simply download script
+`run-fobnail-sdk.sh` either by cloning whole repository or just this one script
+and save it to a directory in your `PATH` environment variable:
+
+```shell
+sudo wget -O /usr/local/bin/run-fobnail-sdk.sh \
+  https://raw.githubusercontent.com/fobnail/fobnail-sdk/main/run-fobnail-sdk.sh
+sudo chmod +x /usr/local/bin/run-fobnail-sdk.sh
+```
+
+With the script installed, one can prepare for the build process itself.
+Firmware can be build for a physical Fobnail Token or simulated on PC, with
+different preparatory steps required. In both cases, first execution of
+`build.sh` downloads Docker container image, which can be both time- and
+bandwidth-consuming.
+
+#### Environment variables common for both targets
+
+Fobnail Token firmware is configured with environment variables passed to
+`build.sh`.
+
+Note that due to the way Docker mounts directories all files and directories
+pointed to by following variables must be located somewhere in `fobnail`
+directory.
+
+- `FOBNAIL_PO_ROOT` - **required** option, must point to valid PEM or DER file
+  with Platform Owner's root certificate. See [this document](/keys_and_certificates/#platform-owner-certificate-chain)
+  for description of PO certificate chain and instructions for building such.
+
+- `FOBNAIL_EK_ROOT_DIR` - points to directory with TPM root certificates.
+  Fobnail repository includes [such directory](https://github.com/fobnail/fobnail/tree/main/tpm_ek_roots)
+  which normally should be used as `FOBNAIL_EK_ROOT_DIR`, but you may change it
+  if certificates supplied in repository became outdated. Either this or
+  `FOBNAIL_EXTRA_EK_ROOT` (or both) **must be specified**, but build process
+  won't fail when neither is passed.
+
+- `FOBNAIL_EXTRA_EK_ROOT` - points to one specific TPM certificate. Useful when
+  testing with TPM emulator, or to limit Fobnail usage to one specific TPM
+  vendor. Either this or `FOBNAIL_EK_ROOT_DIR` (or both) **must be specified**,
+  but build process won't fail when neither is passed.
+
+#### Building and flashing firmware to Fobnail Token
+
+For hardware setup instructions see [Flashing preparation](flashing_preparation.md).
+
+Building and flashing is performed by executing (from `fobnail` directory):
+
+```shell
+env FOBNAIL_PO_ROOT=po_root.crt FOBNAIL_EK_ROOT_DIR=tpm_ek_roots \
+    ./build.sh -t nrf --run
```

Comment:
I'm getting:

```
   Compiling coap-server v0.1.2 (/build/external/coap-server-rs)
   Compiling embassy-nrf v0.1.0 (/build/external/embassy/embassy-nrf)
   Compiling trussed v0.1.0 (/build/external/trussed)
    Finished dev [optimized + debuginfo] target(s) in 1m 49s
warning: unused variable: `state`
  --> src/server/mod.rs:39:5
   |
39 |     state: &ServerState,
   |     ^^^^^ help: if this is intentional, prefix it with an underscore: `_state`
   |
   = note: `#[warn(unused_variables)]` on by default

warning: unused variable: `client`
  --> src/server/mod.rs:40:5
   |
40 |     client: Arc<Mutex<CriticalSectionRawMutex, Client>>,
   |     ^^^^^^ help: if this is intentional, prefix it with an underscore: `_client`

warning: 2 warnings emitted

      Config default
      Target /build/target/thumbv7em-none-eabihf/debug/fobnail
       Error No supported probe was found
```

on fobnail 116e37862a9c

Reply:
looks scary but maybe it is expected when no device was detected?

Reply:
@arturkow2000 ?

Reply:
@krystian-hebel I created PR with fix: https://github.com/fobnail/fobnail/pull/62